### ISSUE TITLE
[BUG]:  For Adam,  schedule with this name not available.

### ISSUE THREAD

**Issue by malosaa** [plugwise/plugwise-beta#276]
### Describe the bug.

Well i have a blueprint, that i later release this year.
After i updated to last version of adam i get this error:

`☀️ Malosa Heater Control ☀️: Choose at step 1: choice 1: Error executing script. Error for call_service at pos 1: Error communicating with API: Plugwise: no schedule with this name available.`

So i double checked the attribute that need to contain 3 names, asleep, home , away.
They are in there,

asleep works, but when it needs home i get this error...

This blueprint worked for almost a year, and now i get an error.
I did even try in capital H, not working 

I quickly tested a bit of my code and does not work, i just added for now to test only a trigger otherwise i could not test,
copy paste it in automations and change to your devices, u see it doesnt work.

Your adam needs to be set on (asleep) before getting this issue.
Its using the anna trough adam, and i have no anna installed into hass

```
alias: test weg
description: ''
trigger:
  - platform: sun
    event: sunset
    offset: 0
condition: []
action:
  - choose:
      - conditions:
          - condition: not
            conditions:
              - condition: state
                entity_id: climate.adam
                attribute: preset_modes
                state: home
        sequence:
          - service: climate.turn_on
            data: {}
            target:
              entity_id: climate.adam
          - service: climate.set_preset_mode
            data:
              preset_mode: home
            target:
              entity_id: climate.adam
    default: []
mode: single
```

### What version of Home Assistant Core has the issue?

2022.4.6

### What was the last working version of Home Assistant Core?

2022.4.6

### What type of installation are you running?

Home Assistant OS

### How did you install plugwise-beta?

HACS

### What kind of Plugwise device is having issues?

Smile: Adam (including Lisa, Tom, Floor)

### What firmware version is your Plugwise product at?

3.6.7

**Comment by malosaa** [comment]
Actually, i thought i fixed it, but still giving a error.

![image](https://user-images.githubusercontent.com/13116501/164723645-8b987ce8-5d68-46ae-b40f-53e78d0c502f.png)

**Comment by malosaa** [comment]
I think you fixed it in the last beta. 

i report back tomorrow.

**Comment by bouwew** [comment]
Hi Hector, I was away, read the issue just now.

Quick feedback: don't mix schedules and presets. The names asleep, home and away are preset names.

We did add a check for the preset-mode name in a recent update, but the names that you enter, look correct.
I just now noticed something strange in our code, I will fix that in version 0.22.3a4.

**Comment by malosaa** [comment]
> Hi Hector, I was away, read the issue just now.
> 
> Quick feedback: don't mix schedules and presets. The names asleep, home and away are preset names.
> 
> We did add a check for the preset-mode name in the climate.set_preset_mode function in a recent update, but the names that you enter, look correct. I just now noticed something strange in our code, I will fix that in version 0.22.3a4. Maybe that will help.

Hi Bouwe,

No worries you guys take your time, no rush.
Yes it is strange, i know its a preset, but home-assistant says schedule.. thats why i opened the ticket :)
It works with the non beta tho, on my test system.

But anyway thanks for replying so fast.

Regards

**Comment by malosaa** [comment]
Hi Bouwe, 

I did download the lest version, still that issue.

On my test system running the version not from hacs and there it works.
I have exact the same blueprint running with the same settings, there is no error.

I don't understand why this happens.

regards

**Comment by malosaa** [comment]
I can confirm its not hass version.

Its a problem in the beta versions, even the last one.
When i try the version not from hacs it works.

Regards

**Comment by bouwew** [comment]
To be able to solve this we need more information:
- Can we look at your blueprint code somewhere?
- Is there any detailed information shown in the HA LOGBOEK when an error occurs?

**Comment by malosaa** [comment]
No there is no error, as now on my test system running with the Adam addon from home assistant, it works normal.
If i install on that same system a version from beta it doesnt work.

The blueprint is the same as my main system, i did not even touch it for months.

this is the only error on my main system when manual activating my automation.
 `Choose at step 1: choice 1: Error executing script. Error for call_service at pos 1: Error communicating with API: Plugwise: no schedule with this name available.
`
when i activate the automation on my test system its working with no error but using a other adam addon.


here is the piece where it checks for the device.
As i have not yet released this blueprint, i need to finish and add some more stuff, but i did not had any time for it

```
state_name_morning:
      description: Here you type the preset-state name of your thermostat when entering the morning and start spending money.
      name: Preset for the morning
      default: "home"
    state_name_night:
      description: Here you type the preset-state name of your thermostat when entering the night time to save some money(lowercase)
      name:  Preset for the night 
      default: "asleep"
```

and here a bit of the action when it enters the morning state,

```
action:
- choose: 
  - conditions: 
    - condition: not
      conditions:
      - condition: state
        entity_id: !input 'heating'
        state: !input 'state_name_morning'
        attribute: preset_mode
    sequence:
    - service: climate.turn_on
      target:
        entity_id: !input 'heating' 
    - service: climate.set_preset_mode
      target:
        entity_id: !input 'heating'
      data:
        preset_mode: !input 'state_name_morning'
```

**Comment by malosaa** [comment]
Try this automation, that one doesnt work for me with the beta 

```
alias: test Adam
description: ''
mode: single
trigger:
  - platform: sun
    event: sunrise
    offset: 0
condition: []
action:
  - choose:
      - conditions:
          - condition: not
            conditions:
              - condition: state
                entity_id: climate.openthermthermostat
                attribute: preset_mode
                state: home
        sequence:
          - service: climate.turn_on
            data: {}
            target:
              entity_id: climate.openthermthermostat
          - service: climate.set_preset_mode
            data:
              preset_mode: home
            target:
              entity_id: climate.openthermthermostat
    default: []
```

**Comment by bouwew** [comment]
What happens when you use: `default: home` without ""?

**Comment by bouwew** [comment]
One more thing: why do you do `climate.turn_on`? Plugwise does not support this command. Also not `climate.turn_off`.

**Comment by malosaa** [comment]
> Update: question: it works in HA Core 2022.4 with Core Plugwise? And it doesn't work with plugwise-beta v0.22.3 added into Core 2022.4?

Yes 

> One more thing: why do you do climate.turn_on? Plugwise does not support this command. Also not climate.turn_off.

I know, but it doesnt give an error, i need to refresh the code still.

**Comment by malosaa** [comment]
@bouwew 

here is the blueprint i created, i just released it.
https://community.home-assistant.io/t/smart-saver-control-v1-0-1/416441

I'm running it for 1 year now.

**Comment by vincentwolsink** [comment]
I am experiencing somewhat similar issues.

When using the `climate.set_preset_mode` service everything works fine.
```
service: climate.set_preset_mode
data:
  preset_mode: asleep
target:
  entity_id: climate.woonkamer
  ```

But when using the climate entity in a scene it returns the same error message.
```
- name: Welterusten
  entities:
    climate.woonkamer:
      state: "heat"
      preset_mode: "asleep"
```
```
2022-05-12 10:51:13 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [139886539396480] Error communicating with API: Plugwise: no schedule with this name available.
Traceback (most recent call last):
  File "/config/custom_components/plugwise/util.py", line 28, in handler
    return await func(self, *args, **kwargs)
  File "/config/custom_components/plugwise/climate.py", line 175, in async_set_hvac_mode
    await self.coordinator.api.set_schedule_state(
  File "/usr/local/lib/python3.9/site-packages/plugwise/smile.py", line 504, in set_schedule_state
    raise PlugwiseError("Plugwise: no schedule with this name available.")
plugwise.exceptions.PlugwiseError: Plugwise: no schedule with this name available.
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 189, in handle_call_service
    await hass.services.async_call(
  File "/usr/src/homeassistant/homeassistant/core.py", line 1627, in async_call
    task.result()
  File "/usr/src/homeassistant/homeassistant/core.py", line 1664, in _execute_service
    await cast(Callable[[ServiceCall], Awaitable[None]], handler.job.target)(
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 204, in handle_service
    await service.entity_service_call(
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 677, in entity_service_call
    future.result()  # pop exception if have
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 964, in async_request_call
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 714, in _handle_entity_call
    await result
  File "/usr/src/homeassistant/homeassistant/components/scene/__init__.py", line 114, in _async_activate
    await self.async_activate(**kwargs)
  File "/usr/src/homeassistant/homeassistant/components/homeassistant/scene.py", line 327, in async_activate
    await async_reproduce_state(
  File "/usr/src/homeassistant/homeassistant/helpers/state.py", line 116, in async_reproduce_state
    await asyncio.gather(
  File "/usr/src/homeassistant/homeassistant/helpers/state.py", line 110, in worker
    await platform.async_reproduce_states(
  File "/usr/src/homeassistant/homeassistant/components/climate/reproduce_state.py", line 85, in async_reproduce_states
    await asyncio.gather(
  File "/usr/src/homeassistant/homeassistant/components/climate/reproduce_state.py", line 52, in _async_reproduce_states
    await call_service(SERVICE_SET_HVAC_MODE, [], {ATTR_HVAC_MODE: state.state})
  File "/usr/src/homeassistant/homeassistant/components/climate/reproduce_state.py", line 47, in call_service
    await hass.services.async_call(
  File "/usr/src/homeassistant/homeassistant/core.py", line 1627, in async_call
    task.result()
  File "/usr/src/homeassistant/homeassistant/core.py", line 1664, in _execute_service
    await cast(Callable[[ServiceCall], Awaitable[None]], handler.job.target)(
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 204, in handle_service
    await service.entity_service_call(
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 677, in entity_service_call
    future.result()  # pop exception if have
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 964, in async_request_call
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 714, in _handle_entity_call
    await result
  File "/config/custom_components/plugwise/util.py", line 30, in handler
    raise HomeAssistantError(
homeassistant.exceptions.HomeAssistantError: Error communicating with API: Plugwise: no schedule with this name available.
```

**Comment by bouwew** [comment]
@vincentwolsink this doesn't work?
```
- name: Welterusten
  entities:
    climate.woonkamer:
      state: heat
      preset_mode: asleep
```

**Comment by vincentwolsink** [comment]
Hi @bouwew,

Tried that too (although it is yaml and shouldn't really matter), but no luck.

**Comment by bouwew** [comment]
Thanks, this error log helps me to pinpoint the problem.
The issue is not with the preset because the error is related to the `set_hvac_mode` command.
This `set_hvac_mode` command calls the plugwise-backend command `set_schedule_state`. And this one needs the last-used schedule name as input. And since you don't have any schedule defined, the above error shows.

**Comment by vincentwolsink** [comment]
I don't think so. Because HA actually enforces me to set a state.

```
- name: Welterusten
  entities:
    climate.woonkamer:
      preset_mode: asleep
```

Results in the following error:
```
2022-05-12 12:23:04 ERROR (MainThread) [homeassistant.config] Invalid config for [scene]: State for climate.woonkamer should be a string for dictionary value @ data['states'][0]['entities']. Got None. (See /config/scenes.yaml, line 13). Please check the docs at https://www.home-assistant.io/integrations/scene
```

**Comment by bouwew** [comment]
I'm digging a bit deeper, looks like HA climate has some new functionality that we hadn't noticed until now.

**Comment by bouwew** [comment]
@malosaa @vincentwolsink please try plugwise-beta v0.23.0a6
This update contains a work-around that should hopefully fix the problem.

**Comment by vincentwolsink** [comment]
v0.23.0a6 fixed it for me. Thanks!

**Comment by bouwew** [comment]
@malosaa Does this update work for you too?

If not, I would suggest removing the indicated line in all the `climate_set_temperature` service calls:
```
    - service: climate.set_temperature
      data:
        entity_id: !input 'heating'
        hvac_mode: heat_cool  <-- remove this line
```

**Comment by bouwew** [comment]
@vincentwolsink @malosaa please test plugwise-beta v0.23.0a7, this version has the final fix for this issue.

**Comment by vincentwolsink** [comment]
Hi @bouwew, v0.23.0a7 also works fine for me!

**Comment by malosaa** [comment]
@bouwew 

sorry for the late reply, i was very sick.. still not 100% better tho.
But the issue has been solved as u said :) 

thanks for it

**Comment by bouwew** [comment]
@malosaa Hope you get back to 100%!

Thanks for reporting back, I'm closing the issue.

**Comment by malosaa** [comment]
@bouwew 

i having this same issue again, its getting cold and wanted to use my blueprint again.
i did try everything before reaching out to you.

im on v0.40.3  plugwise beta/smile/stretch  using older anna and adam.

my error is in this bit of code:

```
action:
- choose:
  - conditions:
    - condition: not
      conditions:
      - condition: state
        entity_id: !input 'heating'
        attribute: preset_mode
        state: !input 'state_name_morning'
    sequence:
    - service: climate.turn_on
      data: {}
      target:
        entity_id: !input 'heating'
    - service: climate.set_preset_mode
      data:
        preset_mode: away
      target:
        entity_id: !input 'heating'
```

however it reads it all ok see this screenshot:
![image](https://github.com/plugwise/plugwise-beta/assets/13116501/ce05248a-1ccf-4e86-a452-513ec8a1f737)

and the trace timeline said it could not connect wit the plugwise api:
![image](https://github.com/plugwise/plugwise-beta/assets/13116501/cdd40a7e-5318-48c2-a756-a9d4e04fe664)


so when executing this exactly same code in developer tool in yaml it executes and sets it to away or home.
![image](https://github.com/plugwise/plugwise-beta/assets/13116501/17b1bfac-8a5f-4f69-b960-724734e05c35)


i could understand if it did not read my variables, but it does.. 

regards

**Comment by bouwew** [comment]
@malosaa I did some digging, the problem is in the `climate.turn_on` call, plugwise-beta does not support that call properly.
Try removing that call from the sequence.

**Comment by bouwew** [comment]
Or, if you feel adventurous, update to v0.43.0a4. That version should provide the proper support (climate.turn_on/_off works in my system running this version.
Note: in v0.43.0a4 the hvac_mode off is added and support for switching the Adam regulation-modes via the climate.set_hvac_mode service.

**Comment by malosaa** [comment]
i did wait till now, and for now i don't see any issue at all.
I'm using it on my main system, so i stay on this version for a bit.

many many thanks and i really appreciate your quick update and work to get this working again 👍

**Comment by bouwew** [comment]
@malosaa please update to v0.43.0a5 and test some more, thanks!
This version contains the definite update for the problem you reported above.

**Comment by malosaa** [comment]
ok ill test now :) 

sorry for the late reply. 
i report back once tested.

**Comment by malosaa** [comment]
i could not reply till now as we had an internet outage from 01:11 till 14:30..
i think its working correctly, but if u want i can wait and test more till tomorrow morning.

regards

**Comment by bouwew** [comment]
Please test the released version: v0.43.0

**Comment by malosaa** [comment]
works perfect :)

**Comment by malosaa** [comment]
Hi @bouwew 

I found an issue with Anna/Adam.
Well now using the heater again the regulation mode doesn't turn off @ adam.
Even when the regulation mode is turned off (show in home assistant) and Anna see the temperature is below the setpoint it still powers on my gas heater
Before when the regulation mode @ adam is turned off it doesn't heat up.

I did try this:
* set Anna from heat to off
* set Adam regulation mode to off 

What does work for now as a work around:
* Set the setpoint to 10c 


What could be the problem ?
Regards
